src/KeyVault/KeyVault/Models/KeyVaultManagementCmdletBase.cs

```diff
@@ -112,9 +112,53 @@ protected List<PSKeyVaultIdentityItem> ListVaults(string resourceGroupName, Hash
                     throw new ArgumentException(PSKeyVaultProperties.Resources.InvalidTagFormat);
                 }
             }
+
+            if (!string.IsNullOrEmpty(tagValuePair.Name))
+            {
+                listResult = listResult.Where(r => r.Tags?.Keys != null && r.Tags.ConvertToDictionary().Keys.Any(k => string.Equals(k, tagValuePair.Name, StringComparison.OrdinalIgnoreCase))).ToList();
+            }
+
+            if (!string.IsNullOrEmpty(tagValuePair.Value))
+            {
+                listResult = listResult.Where(r => r.Tags?.Values != null && r.Tags.ConvertToDictionary().Values.Any(v => string.Equals(v, tagValuePair.Value, StringComparison.OrdinalIgnoreCase))).ToList();
+            }
+
+            return listResult;
+        }
+
+        protected PSKeyVault FilterByTag(PSKeyVault keyvault, Hashtable tag)
```

Comment:
@maddieclayton is this method just checking to see if the provided `PSKeyVault` object has the given tag?

Reply:
Yes - I couldn't think of a good way to have parameter set that accepted either a name or a resourcegroup (but not both) and tags do to filtering, so I added in this extra logic in case you give both, as I would expect the cmdlet to return nothing if the tags of the specific KV don't match.

Reply:
@cormacpayne to prevent duplicating code, we should have this function be implemented as follows:

```cs
protected PSKeyVault FilterByTag(PSKeyVault keyVault, Hashtable tag)
{
    return FilterByTag(new List<PSKeyVaultIdentityItem> { keyVault }, tag).FirstOrDefault();
}
```